gateway-export-plugin/src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/PolicyWriter.java

```diff
@@ -49,28 +51,95 @@ public void write(Bundle bundle, File rootFolder) {
+        services.values().parallelStream().forEach(serviceEntity -> {
+            final String id = serviceEntity.getId();
+            final String name = serviceEntity.getName();
+            final PolicyMetadata policyMetadata = writePolicy(bundle, policyFolder, serviceEntity.getParentFolder().getId(), name, id, serviceEntity.getPolicyXML());
+            final Set<Dependency> policyDependencies = getPolicyDependencies(id, rawBundle);
+            policyMetadata.setUsedEntities(policyDependencies);
+            policyMetadataMap.put(policyMetadata.getNameWithPath(), policyMetadata);
+        });
 
         Stream.of(
                 bundle.getEntities(Policy.class).values().stream(),
                 bundle.getEntities(GlobalPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream(),
                 bundle.getEntities(AuditPolicy.class).values().stream().map(Policy.class::cast).collect(toList()).stream()
         ).flatMap(s -> s)
-                .forEach(policyEntity -> writePolicy(bundle, policyFolder, policyEntity.getParentFolder().getId(), policyEntity.getName(), policyEntity.getPolicyDocument()));
+                .forEach(policyEntity -> {
+                    final String id = policyEntity.getId();
+                    final String name = policyEntity.getName();
+                    final PolicyMetadata policyMetadata = writePolicy(bundle, policyFolder, policyEntity.getParentFolder().getId(), name, id, policyEntity.getPolicyDocument());
+                    final Set<Dependency> policyDependencies = getPolicyDependencies(id, rawBundle);
+                    policyMetadata.setUsedEntities(policyDependencies);
+                    final PolicyType policyType = policyEntity.getPolicyType();
+                    if(policyType != null){
+                        policyMetadata.setType(policyType.getType());
+                    }
+                    policyMetadata.setTag(policyMetadata.getTag());
+                    policyMetadataMap.put(policyMetadata.getNameWithPath(), policyMetadata);
+                });
+        writePolicyMetadata(policyMetadataMap, policyFolder);
+    }
+
+    /**
+     * Writes policy metadata to policy.yml file
+     * @param policyMetadataMap
+     * @param policyFolder
+     */
+    private void writePolicyMetadata(final Map<String, PolicyMetadata> policyMetadataMap, final File policyFolder){
+        if (!policyMetadataMap.isEmpty()) {
+            //build yml file with dependencies
+            ObjectWriter objectWriter = jsonTools.getObjectWriter();
+            File policyMetadataFile = new File(policyFolder, "policies" + jsonTools.getFileExtension());
+            try (OutputStream fileStream = Files.newOutputStream(policyMetadataFile.toPath())) {
+                objectWriter.writeValue(fileStream, policyMetadataMap);
+            } catch (IOException e) {
+                throw new WriteException("Error writing policy metadata", e);
+            }
```

Comment:
This logic can move to `JsonTools`. Similar to `DocumentTools`

Reply:
JsonTools is generic utility for json file operations, its not good idea to refer PolicyMetadata.java in this class

Reply:
handled somepart